ledger/participant-integration-api/src/main/scala/platform/apiserver/configuration/LedgerConfigurationSubscriptionFromIndex.scala

```diff
@@ -44,142 +51,97 @@ private[apiserver] final class LedgerConfigurationSubscriptionFromIndex(
       override def acquire()(implicit
           context: ResourceContext
       ): Resource[LedgerConfigurationSubscription with IsReady] =
-        Resource(Future {
-          new Subscription(configurationLoadTimeout)
-        }(context.executionContext))(subscription =>
+        Resource(
+          indexService
+            .lookupConfiguration()
```

Comment:
What happens if this fails?

Reply:
The resource fails and the participant will crash.

This might sound scary, but this is already the case if the index fails to look up the ledger ID, so it doesn't add any new failure modes. Basically, participant without index is bad news.

Reply:
Worth logging something to clarify or will the cause of failure be clear enough?

Reply:
It's a good point. We generally don't retry database queries or otherwise handle failures on a low level.

For the indexer, we have `RecoveringIndexer` that restarts the whole indexer if anything fails (even a small thing like a transient database connection issue). This is not optimal, but at least the indexer component automatically and recovers from transient errors.

For the ledger API server, (AFAIK) we don't have such a thing. It would be the responsibility of the participant operator to monitor the health of the participant and restart if necessary.

Reply:
If only we could stop packaging the API server and indexer together, that would be trivial. :upside_down_face: